pyincore/analyses/mmsacge/mmsacge.py

```diff
@@ -0,0 +1,169 @@
+        ----------
+        filenames : Dict[str, str]
+            Dictionary containing the factor name and the path to the coefficient file
+
+        Returns
+        -------
+        Tuple[Dict[str, np.ndarray], Dict[str, List[str]]]
+            Tuple containing two dictionaries. The first dictionary contains the model coefficients for each factor and the second dictionary contains the sectors for each factor
+        """
+        model_coeffs: Dict[str, np.ndarray] = {}
+        sectors: Dict[str, List[str]] = {}
+        for factor, filename in filenames.items():
+            logger.info(f"Parsing {filename}")
+            model_coeff_df: pd.DataFrame = pd.read_csv(filename)
+            sectors[factor] = list(model_coeff_df.columns[5:])
+            model_coeffs[factor] = model_coeff_df[
+                model_coeff_df.columns[4:]
+            ].to_numpy()[
+                1:, :
+            ]  # skip the first row as it contains the total value model and its not needed in output
+
+        return model_coeffs, sectors
+
+    def parse_files(
+        self,
+    ) -> Tuple[Dict[str, List[str]], np.ndarray, np.ndarray, Dict[str, np.ndarray]]:
+        """parse_files Utility function to parse the input files
+
+        Returns
+        -------
+        Tuple[Dict[str, List[str]], np.ndarray, np.ndarray, Dict[str, np.ndarray]]
+            Returns a tuple containing the following:
+            1. sectors: Dictionary containing the sectors for each factor
+            2. base_cap_factors: List of numpy arrays containing the base capital for each factor
+            3. base_cap: Numpy array containing the base capital
+            4. model_coeffs: Dictionary containing the model coefficients for each factor
+        """
+        logger.info("Parsing input files...")
+        base_cap_factors, base_cap = self.parse_base_vals(os.path.join(pyglobals.PYINCORE_PACKAGE_HOME, "analyses", "mmsacge", "base_val_shelby_MMSA.xlsx"))
+        model_filenames = {"ds": os.path.join(pyglobals.PYINCORE_PACKAGE_HOME, "analyses", "mmsacge", "DDS_coefficients.csv")}
+        model_coeffs, sectors = self.parse_coeff(model_filenames)
+        logger.info("Parsing input files completed.")
+        
+        return sectors, base_cap_factors, base_cap, model_coeffs
+
+    def run_analysis(self) -> None:
+        logger.info("Running MMSA CGE model...")
+        sector_shocks = pd.read_csv(
+            self.get_input_dataset("sector_shocks").get_file_path("csv")
+        )
+        # arrange the capital shocks in the same order as the sectors
+        shocks = []
+        for sector in self.sectors["ds"]:
+            shocks.append(
+                sector_shocks.loc[sector_shocks["sector"] == sector.upper()]["shock"]
+            )
+        capital_shocks = np.array(shocks).reshape(1, -1)
+        super().run_core_cge_ml(
+            self.base_cap, capital_shocks, self.model_coeffs, self.base_cap_factors[:1] # need to remove this slice for full release
+        )
```

Comment:
Could we pull the " self.set_result_csv_data" out to each analysis. And allow user to append their own custom name to the filename?

Reply:
I don't think I need to pull out the functions, I can try and use the spec parameters and see if they are supplied, I will use them to set the name, otherwise, we can make the function return data frames and let them save it..